vyper/parser/expr.py

```diff
@@ -122,32 +103,21 @@ def integer(self):
                 pos=getpos(self.expr),
             )
 
-    def decimal(self):
+    def parse_Decimal(self):
         numstring, num, den = get_number_as_fraction(self.expr, self.context)
-        # if not SizeLimits.in_bounds('decimal', num // den):
-        # if not SizeLimits.MINDECIMAL * den <= num <= SizeLimits.MAXDECIMAL * den:
         if not (SizeLimits.MINNUM * den <= num <= SizeLimits.MAXNUM * den):
-            raise InvalidLiteral("Number out of range: " + numstring, self.expr)
+            return
         if DECIMAL_DIVISOR % den:
-            raise InvalidLiteral(
-                "Type 'decimal' has maximum 10 decimal places",
-                self.expr
-            )
+            return
```

Comment:
Why return `None`?

Reply:
I'm using a pattern where a check is made on the return value, and if `None`, an internal error is raised. This reduces the number of unreachable lines of code while still providing a layer of safety in case type checking is insufficient.